lib/test-requirements.txt

```diff
@@ -19,3 +19,4 @@ setuptools<50.0.0
 sqlalchemy>=1.4.25
 watchdog==2.1.5
 rich>=11.2.0
+scipy
```

Comment:
In keeping with the pattern of other requirements, perhaps we ought to specify a version? Latest scipy version is `1.9.1`, but `1.7` is the latest that supports python 3.7.  Version `1.4.1` is what we previously required. I'm assuming there is no harm in using version `<=1.7`.

Also super nit - could we put this requirement after line 16 (`pyodbc>=4.0.32`) but before 17 (`seaborn>=0.11.2`) to keep the existing alphabetical order?